src/modules/navigator/mission_block.cpp

```diff
@@ -432,37 +432,34 @@ MissionBlock::is_mission_item_reached()
 			const float dist_current_next = get_distance_to_next_waypoint(curr_sp_new->lat, curr_sp_new->lon, next_sp.lat,
 							next_sp.lon);
 
-			/* enforce exit heading if in FW, the next wp is valid, the vehicle is currently loitering and either having force_heading set,
+			/* enforce exit course if in FW, the next wp is valid, the vehicle is currently loitering and either having force_heading set,
 			   or if loitering to achieve altitdue at a NAV_CMD_WAYPOINT */
-			const bool enforce_exit_heading = _navigator->get_vstatus()->vehicle_type != vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+			const bool enforce_exit_course = _navigator->get_vstatus()->vehicle_type != vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
 							  &&
 							  next_sp.valid &&
 							  curr_sp_new->type == position_setpoint_s::SETPOINT_TYPE_LOITER &&
 							  (_mission_item.force_heading || _mission_item.nav_cmd == NAV_CMD_WAYPOINT);
 
-			// can only enforce exit heading if next waypoint is not within loiter radius of current waypoint
-			const bool exit_heading_is_reachable = dist_current_next > 1.2f * curr_sp_new->loiter_radius;
+			// can only enforce exit course if next waypoint is not within loiter radius of current waypoint
+			const bool exit_course_is_reachable = dist_current_next > 1.2f * curr_sp_new->loiter_radius;
 
-			if (enforce_exit_heading && exit_heading_is_reachable) {
-
-				float yaw_err = 0.0f;
+			if (enforce_exit_course && exit_course_is_reachable) {
 
 				// set required yaw from bearing to the next mission item
-				_mission_item.yaw = get_bearing_to_next_waypoint(_navigator->get_global_position()->lat,
-						    _navigator->get_global_position()->lon,
-						    next_sp.lat, next_sp.lon);
+				const float desired_exit_bearing = get_bearing_to_next_waypoint(_navigator->get_global_position()->lat,
+								   _navigator->get_global_position()->lon,
+								   next_sp.lat, next_sp.lon);
 				const float cog = atan2f(_navigator->get_local_position()->vy, _navigator->get_local_position()->vx);
-				yaw_err = wrap_pi(_mission_item.yaw - cog);
 
-				exit_heading_reached = fabsf(yaw_err) < _navigator->get_yaw_threshold();
+				exit_course_reached = fabsf(wrap_pi(desired_exit_bearing - cog)) < kLoiterExitCourseReachedThreshold;
```

Comment:
while we're modifying things in here - can we also make `cog` less ambiguous? e.g. `vehicle_course`? a passerby in the code may see cog and say.. "what? center of gravity?"

Another option.. could make this a vector operation and save the atan2 cost, e.g. using 
https://github.com/PX4/PX4-Autopilot/blob/40149ecabcf33238c6e51708d00412fbd186b03c/src/lib/geo/geo.h#L128-L132
something like:
```cpp
float vehicle_position_to_next_waypoint_north;
float vehicle_position_to_next_waypoint_east;
get_vector_to_next_waypoint(_navigator->get_global_position()->lat, _navigator->get_global_position()->lon, next_sp.lat, next_sp.lon, &v_n, &vehicle_position_to_next_waypoint_north,  &vehicle_position_to_next_waypoint_east);

// this vector defines the exit bearing
const matrix::Vector2f vector_vehicle_position_to_next_waypoint = {vehicle_position_to_next_waypoint_north, vehicle_position_to_next_waypoint_east);

const matrix::Vector2f vehicle_ground_velocity = {_navigator->get_local_position()->vx, _navigator->get_local_position()->vy};
const float ground_speed = vehicle_ground_velocity.norm();

exit_course_reached = (ground_speed > kLoiterExitGroundSpeedThreshold) && (vector_vehicle_position_to_next_waypoint.dot(vehicle_ground_velocity) > vector_vehicle_position_to_next_waypoint.norm() * ground_speed * kCosineExitCourseThreshold);
```

Reply:
@tstastny addressed it like that. As discussed I left away the additional check for min groundspeed though, don't think this is required.